### PR TITLE
DSA-33-calculate-the-height-of-a-binary-tree-iterative-and-recursive

### PR DESCRIPTION
**Pull request by charan-2316**
It Calculates the height of a binary tree – Iterative and RecursiveWrite an efficient algorithm to compute the binary tree’s height

### CODE REVIEW

File: DSA_Training/BinaryTreeHeight.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import java.util.LinkedList;
```
Comment (ganeshyevle) [inline DSA_Training/BinaryTreeHeight.java:1]: Please, add here proper comments

File: DSA_Training/BinaryTreeHeight.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import java.util.Queue;
+import java.util.Scanner;
+
+class TreeNode {
```
Comment (ganeshyevle) [inline DSA_Training/BinaryTreeHeight.java:5]: Please, add here proper comments related to this class

File: DSA_Training/BinaryTreeHeight.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    int value;
+    TreeNode left, right;
+
```
Comment (ganeshyevle) [inline DSA_Training/BinaryTreeHeight.java]: Please, remove this space

File: DSA_Training/BinaryTreeHeight.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    // Iterative method to calculate height
+    int heightIterative(TreeNode root) {
+        if (root == null)
+            return 0;
+
+        Queue<TreeNode> queue = new LinkedList<>();
+        queue.add(root);
+        int height = 0;
+
+        while (!queue.isEmpty()) {
```
Comment (ganeshyevle) [inline DSA_Training/BinaryTreeHeight.java:32]: Please, add here proper comments

File: DSA_Training/BinaryTreeHeight.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        int n = scanner.nextInt();
+        int[] nodes = new int[n];
+
+        System.out.println("Enter the values of nodes (use -1 for null nodes):");
+        for (int i = 0; i < n; i++) {
+            nodes[i] = scanner.nextInt();
+        }
+
+        tree.root = tree.buildTreeFromLevelOrder(nodes);
+
```
Comment (ganeshyevle) [inline DSA_Training/BinaryTreeHeight.java]: Please, remove this pace

**Review by ganeshyevle — CHANGES_REQUESTED** [review]
Do changes that Given

File: DSA_Training/BinaryTreeHeight.java (hunk 1 above)
Comment (charan-2316) [reply DSA_Training/BinaryTreeHeight.java:1]: done

File: DSA_Training/BinaryTreeHeight.java (hunk 2 above)
Comment (charan-2316) [reply DSA_Training/BinaryTreeHeight.java:5]: done

File: DSA_Training/BinaryTreeHeight.java (hunk 3 above)
Comment (charan-2316) [reply DSA_Training/BinaryTreeHeight.java]: done

File: DSA_Training/BinaryTreeHeight.java (hunk 4 above)
Comment (charan-2316) [reply DSA_Training/BinaryTreeHeight.java:32]: done

**Review by charan-2316 — COMMENTED** [review]
done

**Review by ganeshyevle — APPROVED** [review]
All done

**Comment by ganeshyevle** [comment]
Now you can merge

**Pull request merged**